src/oc_zipkin_reporter.erl

```diff
@@ -0,0 +1,101 @@
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% @doc Exports spans to Zipkin
+%% @end
+%%%------------------------------------------------------------------------
+
+
+-module(oc_zipkin_reporter).
+
+-include("opencensus.hrl").
+
+-behaviour(oc_reporter).
+
+-define(DEFAULT_ZIPKIN_ADDRESS, "http://localhost:9411").
+-define(DEFAULT_LOCAL_ENDPOINT, #{<<"serviceName">> => node()}).
+
+-export([init/1,
+         report/2]).
+
+init(Opts) ->
+    Opts.
+
+report(Spans, _Opts) ->
+    Address = zipkin_address(),
+    ZSpans = [zipkin_span(Span) || Span <- Spans],
+
+    try jsx:encode(ZSpans) of
+        JSON ->
+            case httpc:request(post,
+                               {Address ++ "/api/v2/spans", [],
+                                "application/json",
+                                JSON
+                               }, [], []) of
+                {ok, {{_, 202, _}, _, _}} ->
+                    ok;
+                {ok, {{_, Code, _}, _, Message}} ->
+                    report_error("Unable to send spans, Zipkin reported an error: ~p : ~p", [Code, Message]);
+
+                {error, Reason} ->
+                    report_error("Unable to send spans, client error: ~p", [Reason])
+            end
+    catch
+        error:Error ->
+            report_error("Can't spans encode to json: ~p", [Error])
+    end.
+
+zipkin_span(Span) ->
+
+    ParentId = case Span#span.parent_span_id of
+                   undefined -> null;
+                   Id -> iolist_to_binary(io_lib:format("~16.16.0b", [Id]))
+               end,
+    #{
+       <<"traceId">> => iolist_to_binary(io_lib:format("~32.16.0b", [Span#span.trace_id])),
+       <<"name">> => iolist_to_binary(Span#span.name),
+       <<"parentId">> => ParentId,
+       <<"id">> => iolist_to_binary(io_lib:format("~16.16.0b", [Span#span.span_id])),
+       <<"kind">> => <<"SERVER">>, %% TODO: get from attributes?
+       <<"timestamp">> => wts:to_absolute(Span#span.start_time),
+       <<"duration">> => wts:duration(Span#span.start_time, Span#span.end_time),
+       <<"debug">> => false, %% TODO: get from attributes?
+       <<"shared">> => false, %% TODO: get from attributes?
+       <<"localEndpoint">> => local_endpoint(),
+       %% <<"remoteEndpoint">> =>  %% TODO: get from attributes?
+       <<"annotations">> => [],
+       <<"tags">> => to_tags(Span#span.attributes) %% TODO: merge with oc_tags?
+     }.
+
+to_tag(_Name, Value) when is_function(Value) ->
+    Value();
+to_tag(_Name, Value) ->
+    Value.
+
+to_tags(Attributes) ->
+    maps:map(fun(Name, Value) ->
+                     to_tag(Name, Value)
+             end, Attributes).
+
+report_error(Message, Reason) ->
+    erlang:error({zipkin_error, Message,  Reason}).
```

Comment:
Maybe we need an abstraction so we don't have to duplicate the logic in each reporter, but esp for an http reporter it should retry a couple times after an error instead of just giving up on the first failure. 

And not sure if we want to not just report an error with `error_reporter` instead of killing the reporter process.

Reply:
what is error reporter?

Reply:
not sure how to implement retries (hence they are not there). When spans reach reporter they are gone OTOH after reporting interval reporter will be called again, and have to cancel previous attempts otherwise looks like DOS.

Reply:
Oops, meant `error_logger`.

In the google reporter I just do: https://github.com/tsloughter/oc_google_reporter/blob/master/src/oc_google_reporter_v2.erl#L54

It also doesn't have retries yet, but I figured with some backoff it would be ok, the reporter won't get called again until interval time after returning, so there will always be a delay and not a flood of report attempts at once.

First we have to decide on the best way to handle errors, whether to crash or simply log.

Reply:
oc_reporter wraps report call in try catch. So nothing really crash, but log is good. I'll add

Reply:
Oh, right, forgot about that :)